crates/weaver_forge/src/config.rs

```diff
@@ -306,13 +211,30 @@ impl Default for TemplateSyntax {
 pub struct WhitespaceControl {
     /// Configures the behavior of the first newline after a block.
     /// See <https://docs.rs/minijinja/latest/minijinja/struct.Environment.html#method.set_trim_blocks>
-    pub trim_blocks: bool,
+    pub trim_blocks: Option<bool>,
     /// Configures the behavior of leading spaces and tabs from the start of a line to a block.
     /// See <https://docs.rs/minijinja/latest/minijinja/struct.Environment.html#method.set_lstrip_blocks>
-    pub lstrip_blocks: bool,
+    pub lstrip_blocks: Option<bool>,
     /// Configures whether trailing newline are preserved when rendering templates.
     /// See <https://docs.rs/minijinja/latest/minijinja/struct.Environment.html#method.set_keep_trailing_newline>
-    pub keep_trailing_newline: bool,
+    pub keep_trailing_newline: Option<bool>,
+}
+
+impl WhitespaceControl {
+    /// Override the current `WhitespaceControl` with the `WhitespaceControl` passed as argument.
+    /// The merge is done in place. The `WhitespaceControl` passed as argument will be consumed and
+    /// used to override the current `WhitespaceControl`.
+    pub fn override_with(&mut self, other: WhitespaceControl) {
```

Comment:
It's interesting you chose this manner of override. I consider this right to left override.

E.g.

```rust
let mut generic_config = load_generic();
let specific_config = load_specific();
generic_config.override_with(specific_config);
```

I haven't seen the directory walking yet, but I assume this is easier to understand / "fold" the config in this manner?

Reply:
I feel like your example is identical to what is done in this PR. The `other` parameter in the `override_with` is actually a more specific configuration, just like in your example. Maybe there is something ambiguous somewhere in my comments that led you to think otherwise, or perhaps I misinterpreted your example.